### DISCUSSION TITLE
Unawaited async function throws an error and crashes app

### DISCUSSION THREAD

**Discussion by paymog** [fastify/fastify#5063] [Q&A] [answered]
I'm running fastify v3. In one of my paths, I call an async function without `await`ing it kinda like so.

```ts
  fastify.put<SomeType>(
    "mypath",
    {
      schema: SomeSchema,
    },
    async (req, reply) => {
      // stuff

      await deploySubgraph({});

      reply.send({
        data: 'stuff'
      });
    }
  );

export async function deploySubgraph(args: {}) {
  //  lots of stuff

  // this is an async function which throws sometimes
  hubspotClient.userDeployedSubgraph(user.email);

  return 'something useful';
}
```

I'm finding that sometimes `hubspotClient.userDeployedSubgraph` throws an error. I'm surprised that this error is causing my server to crash even though the promise isn't awaited. Is that expected behavior?

I'm specifically seeing this when running my server with nodemon - could nodemon be the issue here?

```
/Users/paymahn/code/goldsky/graph-tools/packages/core/src/hubspot/client.ts:71
      throw new Error("Could not create contact");
            ^
Error: Could not create contact
    at HubspotClient.createContactIfNotExists (/Users/paymahn/code/goldsky/graph-tools/packages/core/src/hubspot/client.ts:71:13)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async HubspotClient.userDeployedSubgraph (/Users/paymahn/code/goldsky/graph-tools/packages/core/src/hubspot/client.ts:94:21)
[nodemon] app crashed - waiting for file changes before starting...
```

**Comment by mcollina** [comment, marked as answer]
A few notes:

1. fastify v3 is out of support, update to v4
2. you are not awaiting your route. It's a Node.js best practice to crash your application if there is an unhandled error. Fastify has nothing to do with this.
3. if you think your function can throw, do `ubspotClient.userDeployedSubgraph(user.email).catch((err) => req.log.warn({ err }))`

**Reply by paymog** [reply]
Thanks for the super fast response! Yeah, we're upgrading to v4 as we speak, we have an open PR internally for the upgrade, haven't finalized it yet though. 

Understood that this has nothing to do with fastify. That being said, would the semantics be different if I *did* await my route?

And yeah, good call. Will add a `.catch` to this hubspot call.

**Reply by mcollina** [reply]
> Understood that this has nothing to do with fastify. That being said, would the semantics be different if I did await my route?

Yes, the error would be caught and reported as a 500 to the caller.
